joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -60,8 +61,51 @@ trait AstForTypesCreator(implicit withSchemaValidation: ValidationMode) { this:
       inherits = inheritsFrom,
       alias = None
     )
-    scope.surroundingAstLabel.foreach(typeDecl.astParentType(_))
-    scope.surroundingScopeFullName.foreach(typeDecl.astParentFullName(_))
+
+    /** Pushes new NamespaceScope onto scope stack and populates AST_PARENT_FULL_NAME and AST_PARENT_TYPE for TypeDecls
+      * that are declared in a namespace
+      * @param typeDecl
+      *   \- TypeDecl node
+      * @param astParentFullName
+      *   \- Fullname of AstParent
+      * @return
+      *   typeDecl node with updated fields
+      */
+    def populateAstParentValues(typeDecl: NewTypeDecl, astParentFullName: String): NewTypeDecl = {
+      val namespaceBlockFullName = s"${scope.surroundingScopeFullName.getOrElse("")}.${astParentFullName}"
+      scope.pushNewScope(NamespaceScope(s"${scope.surroundingScopeFullName.getOrElse("")}.${astParentFullName}"))
+
+      val namespaceBlock =
+        NewNamespaceBlock().name(astParentFullName).fullName(astParentFullName).filename(relativeFileName)
+
+      diffGraph.addNode(namespaceBlock)
+
+      scope.findFileNode() match {
+        case Some(fileNode) => diffGraph.addEdge(fileNode, namespaceBlock, EdgeTypes.AST)
+        case None           => // do nothing
+      }
+
+      typeDecl.astParentFullName(astParentFullName)
+      typeDecl.astParentType(NodeTypes.NAMESPACE_BLOCK)
+
+      typeDeclTemp.fullName(computeFullName(className))
+      typeDecl
+    }
+
+    val (typeDecl, shouldPopScope) = node match {
```

Comment:
Just a suggestion to be more descriptive
```suggestion
    val (typeDecl, shouldPopAdditionalScope) = node match {
```